src/components/ActionButton/ToggleActionButton.ts

```diff
@@ -0,0 +1,182 @@
+         * ```
+         */
+        tooltip: ComponentOptions.buildStringOption(),
+
+        /**
+         * Specifies the handler called when the button is clicked.
+         *
+         * Default is `null`.
+         *
+         * This option is set in JavaScript when initializing the component.
+         */
+        click: ComponentOptions.buildCustomOption(s => null),
+
+        /**
+         * Specifies the handler called when the button is activated.
+         *
+         * Default is `null`.
+         *
+         * This option is set in JavaScript when initializing the component.
+         */
+        activate: ComponentOptions.buildCustomOption(s => null),
+
+        /**
+         * Specifies the handler called when the button is deactivated.
+         *
+         * Default is `null`.
+         *
+         * This option is set in JavaScript when initializing the component.
+         */
+        deactivate: ComponentOptions.buildCustomOption(s => null)
+    };
+
+    private _isActivated: boolean = false;
+    private innerButton: ActionButton;
+
+    constructor(public element: HTMLElement, public options: IToggleActionButtonOptions, public bindings?: IResultsComponentBindings) {
+        super(element, ToggleActionButton.ID, bindings);
+        this.options = ComponentOptions.initComponentOptions(element, ToggleActionButton, options);
+
+        this.createInnerButton(bindings);
+    }
+
+    /**
+     * Indicates whether the toggle button is in the activated state.
+     */
+    public isActivated(): boolean {
+        return this._isActivated;
+    }
+
+    /**
+     * Sets the toggle button to the specified state.
+     * @param activated Whether the button is activated.
+     */
+    public setActivated(activated: boolean): void {
+        if (activated !== this.isActivated()) {
+            this._isActivated = activated;
+            this.updateButton();
+
+            if (this._isActivated && this.options.activate) {
+                this.options.activate();
+            }
+            if (!this._isActivated && this.options.deactivate) {
+                this.options.deactivate();
+            }
+        }
+    }
+
+    protected onClick(): void {
+        this.setActivated(!this.isActivated());
+
+        if (this.options.click) {
+            this.options.click();
+        }
+    }
+
+    private createInnerButton(bindings?: IResultsComponentBindings): void {
+        this.innerButton = new ActionButton(
+            this.element,
+            {
+                icon: this.options.icon,
+                tooltip: this.options.tooltip,
+                click: () => this.onClick()
+            },
+            bindings
+        );
+
+        this.updateButton();
+    }
+
+    private updateButton() {
+        if (this._isActivated) {
+            this.element.classList.add(ToggleActionButton.ACTIVATED_CLASS_NAME);
+            this.element.setAttribute('aria-pressed', 'true');
```

Comment:
TIL: If you want a button to be considered a toggle for ARIA, the attribute need to be set no matter what.
You cannot use the shortcut of just having the attribute.

Reply:
Yes, but the attribute is always set. The constructor calls `createInnerButton`, which in turn calls `updateButton`. So, the `aria-pressed` attribute is set right from the first rendering.